### PR TITLE
GEOMESA-1995 Resolving classpath issues with spark clearspring classes

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Copied clearspring implementations for CountMinSketch, HyperLogLog and StreamSummary (in scala)
* Exposed underlying data structures for fast serialization, without storing class names
* Removed RichCountMinSketch to avoid classloader issues with package protected access
* Removed relocation of clearspring in spark-runtime jars

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/StatsCombiner.scala [hunk 1]
```diff
@@ -44,27 +44,40 @@ class StatsCombiner extends Combiner with LazyLogging {
...
+          return head // couldn't parse anything... return head value and let client deal with it
+        }
+      }
+
       iter.foreach { s =>
-        try {
-          stat += serializer.deserialize(s.get)
-        } catch {
-          case e: Exception => logger.error("Error combining stats:", e)
+        try { stat += serializer.deserialize(s.get) } catch {
```
Comment (aheyne) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/StatsCombiner.scala]: Did you intend to use the new deserialize function here?

Comment (aheyne) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/StatsCombiner.scala]: Did you intend to use the new deserialize function here?

Comment (elahrvivaz) [reply geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/StatsCombiner.scala]: no, we'd still need a try/catch around the `+=`, and we'd have to check for null. it's a bit redundant, but was trying not to have extra checks

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/clearspring/CountMinSketch.scala [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  *
+  * This product includes software developed by AddThis.
+  *
+  * This product also includes code adapted from:
+  *
+  * Apache Solr (http://lucene.apache.org/solr/)
+  * Copyright 2014 The Apache Software Foundation
+  *
+  * Apache Mahout (http://mahout.apache.org/)
+  * Copyright 2014 The Apache Software Foundation
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/clearspring/CountMinSketch.scala]: I think this requires a CQ.:(

Comment (elahrvivaz) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/clearspring/CountMinSketch.scala]: I created a CQ here: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14157
Used 'third party code request' but marked as modified and both source and binary

**Pull request merged**